publishReleaseBundle broken in Android Studio 3.6 Canary 2

### Describe the bug

FAILURE: Build failed with an exception.

### How To Reproduce
Try to publish bundle with AS 3.6 Alpha 2

#### Versions

Plugin v2.2.1
Gradle 5.4
AS 3.6 Alpha 2

### Additional context (if a crash, provide stack trace)
```
FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':app:publishReleaseBundle'.
> Collection is empty.

* Try:
Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.api.internal.tasks.TaskDependencyResolveException: Could not determine the dependencies of task ':app:publishReleaseBundle'.
	at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext.getDependencies(CachingTaskDependencyResolveContext.java:68)
	at org.gradle.execution.plan.TaskDependencyResolver.resolveDependenciesFor(TaskDependencyResolver.java:46)
	at org.gradle.execution.plan.LocalTaskNode.getDependencies(LocalTaskNode.java:111)
	at org.gradle.execution.plan.LocalTaskNode.resolveDependencies(LocalTaskNode.java:79)
	at org.gradle.execution.plan.DefaultExecutionPlan.addEntryTasks(DefaultExecutionPlan.java:177)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph.addEntryTasks(DefaultTaskExecutionGraph.java:139)
	at org.gradle.execution.TaskNameResolvingBuildConfigurationAction.configure(TaskNameResolvingBuildConfigurationAction.java:48)
	at org.gradle.execution.DefaultBuildConfigurationActionExecuter.configure(DefaultBuildConfigurationActionExecuter.java:57)
	at org.gradle.execution.DefaultBuildConfigurationActionExecuter.access$200(DefaultBuildConfigurationActionExecuter.java:26)
	at org.gradle.execution.DefaultBuildConfigurationActionExecuter$2.proceed(DefaultBuildConfigurationActionExecuter.java:63)
	at org.gradle.execution.DefaultTasksBuildExecutionAction.configure(DefaultTasksBuildExecutionAction.java:44)
	at org.gradle.execution.DefaultBuildConfigurationActionExecuter.configure(DefaultBuildConfigurationActionExecuter.java:57)
	at org.gradle.execution.DefaultBuildConfigurationActionExecuter.access$200(DefaultBuildConfigurationActionExecuter.java:26)
	at org.gradle.execution.DefaultBuildConfigurationActionExecuter$2.proceed(DefaultBuildConfigurationActionExecuter.java:63)
	at org.gradle.execution.ExcludedTaskFilteringBuildConfigurationAction.configure(ExcludedTaskFilteringBuildConfigurationAction.java:47)
	at org.gradle.execution.DefaultBuildConfigurationActionExecuter.configure(DefaultBuildConfigurationActionExecuter.java:57)
	at org.gradle.execution.DefaultBuildConfigurationActionExecuter.access$200(DefaultBuildConfigurationActionExecuter.java:26)
	at org.gradle.execution.DefaultBuildConfigurationActionExecuter$1.run(DefaultBuildConfigurationActionExecuter.java:43)
	at org.gradle.internal.Factories$1.create(Factories.java:25)
	at org.gradle.api.internal.project.DefaultProjectStateRegistry.withLenientState(DefaultProjectStateRegistry.java:132)
	at org.gradle.api.internal.project.DefaultProjectStateRegistry.withLenientState(DefaultProjectStateRegistry.java:124)
	at org.gradle.execution.DefaultBuildConfigurationActionExecuter.select(DefaultBuildConfigurationActionExecuter.java:39)
	at org.gradle.initialization.DefaultGradleLauncher$CalculateTaskGraph.run(DefaultGradleLauncher.java:333)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:402)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:394)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:92)
	at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
	at org.gradle.initialization.DefaultGradleLauncher.constructTaskGraph(DefaultGradleLauncher.java:218)
	at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:155)
	at org.gradle.initialization.DefaultGradleLauncher.executeTasks(DefaultGradleLauncher.java:134)
	at org.gradle.internal.invocation.GradleBuildController$1.execute(GradleBuildController.java:58)
	at org.gradle.internal.invocation.GradleBuildController$1.execute(GradleBuildController.java:55)
	at org.gradle.internal.invocation.GradleBuildController$3.create(GradleBuildController.java:82)
	at org.gradle.internal.invocation.GradleBuildController$3.create(GradleBuildController.java:75)
	at org.gradle.internal.work.DefaultWorkerLeaseService.withLocks(DefaultWorkerLeaseService.java:183)
	at org.gradle.internal.work.StopShieldingWorkerLeaseService.withLocks(StopShieldingWorkerLeaseService.java:40)
	at org.gradle.internal.invocation.GradleBuildController.doBuild(GradleBuildController.java:75)
	at org.gradle.internal.invocation.GradleBuildController.run(GradleBuildController.java:55)
	at org.gradle.tooling.internal.provider.runner.BuildModelActionRunner.run(BuildModelActionRunner.java:54)
	at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
	at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
	at org.gradle.launcher.exec.BuildOutcomeReportingBuildActionRunner.run(BuildOutcomeReportingBuildActionRunner.java:58)
	at org.gradle.tooling.internal.provider.ValidatingBuildActionRunner.run(ValidatingBuildActionRunner.java:32)
	at org.gradle.launcher.exec.BuildCompletionNotifyingBuildActionRunner.run(BuildCompletionNotifyingBuildActionRunner.java:39)
	at org.gradle.launcher.exec.RunAsBuildOperationBuildActionRunner$3.call(RunAsBuildOperationBuildActionRunner.java:51)
	at org.gradle.launcher.exec.RunAsBuildOperationBuildActionRunner$3.call(RunAsBuildOperationBuildActionRunner.java:45)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:416)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:406)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:102)
	at org.gradle.internal.operations.DelegatingBuildOperationExecutor.call(DelegatingBuildOperationExecutor.java:36)
	at org.gradle.launcher.exec.RunAsBuildOperationBuildActionRunner.run(RunAsBuildOperationBuildActionRunner.java:45)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter$1.transform(InProcessBuildActionExecuter.java:49)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter$1.transform(InProcessBuildActionExecuter.java:46)
	at org.gradle.composite.internal.DefaultRootBuildState.run(DefaultRootBuildState.java:78)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:46)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:31)
	at org.gradle.launcher.exec.BuildTreeScopeBuildActionExecuter.execute(BuildTreeScopeBuildActionExecuter.java:42)
	at org.gradle.launcher.exec.BuildTreeScopeBuildActionExecuter.execute(BuildTreeScopeBuildActionExecuter.java:28)
	at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:78)
	at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:52)
	at org.gradle.tooling.internal.provider.SubscribableBuildActionExecuter.execute(SubscribableBuildActionExecuter.java:59)
	at org.gradle.tooling.internal.provider.SubscribableBuildActionExecuter.execute(SubscribableBuildActionExecuter.java:36)
	at org.gradle.tooling.internal.provider.SessionScopeBuildActionExecuter.execute(SessionScopeBuildActionExecuter.java:68)
	at org.gradle.tooling.internal.provider.SessionScopeBuildActionExecuter.execute(SessionScopeBuildActionExecuter.java:38)
	at org.gradle.tooling.internal.provider.GradleThreadBuildActionExecuter.execute(GradleThreadBuildActionExecuter.java:37)
	at org.gradle.tooling.internal.provider.GradleThreadBuildActionExecuter.execute(GradleThreadBuildActionExecuter.java:26)
	at org.gradle.tooling.internal.provider.ParallelismConfigurationBuildActionExecuter.execute(ParallelismConfigurationBuildActionExecuter.java:43)
	at org.gradle.tooling.internal.provider.ParallelismConfigurationBuildActionExecuter.execute(ParallelismConfigurationBuildActionExecuter.java:29)
	at org.gradle.tooling.internal.provider.StartParamsValidatingActionExecuter.execute(StartParamsValidatingActionExecuter.java:60)
	at org.gradle.tooling.internal.provider.StartParamsValidatingActionExecuter.execute(StartParamsValidatingActionExecuter.java:32)
	at org.gradle.tooling.internal.provider.SessionFailureReportingActionExecuter.execute(SessionFailureReportingActionExecuter.java:55)
	at org.gradle.tooling.internal.provider.SessionFailureReportingActionExecuter.execute(SessionFailureReportingActionExecuter.java:41)
	at org.gradle.tooling.internal.provider.SetupLoggingActionExecuter.execute(SetupLoggingActionExecuter.java:48)
	at org.gradle.tooling.internal.provider.SetupLoggingActionExecuter.execute(SetupLoggingActionExecuter.java:32)
	at org.gradle.launcher.daemon.server.exec.ExecuteBuild.doBuild(ExecuteBuild.java:67)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
	at org.gradle.launcher.daemon.server.exec.WatchForDisconnection.execute(WatchForDisconnection.java:37)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
	at org.gradle.launcher.daemon.server.exec.ResetDeprecationLogger.execute(ResetDeprecationLogger.java:26)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
	at org.gradle.launcher.daemon.server.exec.RequestStopIfSingleUsedDaemon.execute(RequestStopIfSingleUsedDaemon.java:34)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
	at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:74)
	at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:72)
	at org.gradle.util.Swapper.swap(Swapper.java:38)
	at org.gradle.launcher.daemon.server.exec.ForwardClientInput.execute(ForwardClientInput.java:72)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
	at org.gradle.launcher.daemon.server.exec.LogAndCheckHealth.execute(LogAndCheckHealth.java:55)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
	at org.gradle.launcher.daemon.server.exec.LogToClient.doBuild(LogToClient.java:62)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
	at org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment.doBuild(EstablishBuildEnvironment.java:81)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
	at org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy$1.run(StartBuildOrRespondWithBusy.java:50)
	at org.gradle.launcher.daemon.server.DaemonStateCoordinator$1.run(DaemonStateCoordinator.java:295)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: java.util.NoSuchElementException: Collection is empty.
	at kotlin.collections.CollectionsKt___CollectionsKt.single(_Collections.kt:501)
	at com.github.triplet.gradle.play.tasks.PublishBundle$bundle$2.invoke(PublishBundle.kt:30)
	at com.github.triplet.gradle.play.tasks.PublishBundle$bundle$2.invoke(PublishBundle.kt:21)
	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
	at com.github.triplet.gradle.play.tasks.PublishBundle.getBundle(PublishBundle.kt)
	at org.gradle.api.internal.tasks.properties.bean.AbstractNestedRuntimeBeanNode$BeanPropertyValue$1$1.create(AbstractNestedRuntimeBeanNode.java:75)
	at org.gradle.util.SingleMessageLogger.whileDisabled(SingleMessageLogger.java:341)
	at org.gradle.api.internal.tasks.properties.bean.AbstractNestedRuntimeBeanNode$BeanPropertyValue$1.get(AbstractNestedRuntimeBeanNode.java:72)
	at com.google.common.base.Suppliers$NonSerializableMemoizingSupplier.get(Suppliers.java:167)
	at org.gradle.api.internal.tasks.properties.bean.AbstractNestedRuntimeBeanNode$BeanPropertyValue.call(AbstractNestedRuntimeBeanNode.java:119)
	at org.gradle.util.GUtil.uncheckedCall(GUtil.java:459)
	at org.gradle.api.internal.file.collections.BuildDependenciesOnlyFileCollectionResolveContext.add(BuildDependenciesOnlyFileCollectionResolveContext.java:64)
	at org.gradle.api.internal.file.collections.BuildDependenciesOnlyFileCollectionResolveContext.add(BuildDependenciesOnlyFileCollectionResolveContext.java:72)
	at org.gradle.api.internal.file.DefaultFileCollectionFactory$ResolvingFileCollection.visitContents(DefaultFileCollectionFactory.java:209)
	at org.gradle.api.internal.file.CompositeFileCollection.visitDependencies(CompositeFileCollection.java:186)
	at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext$TaskGraphImpl.getNodeValues(CachingTaskDependencyResolveContext.java:106)
	at org.gradle.internal.graph.CachingDirectedGraphWalker$GraphWithEmptyEdges.getNodeValues(CachingDirectedGraphWalker.java:211)
	at org.gradle.internal.graph.CachingDirectedGraphWalker.doSearch(CachingDirectedGraphWalker.java:121)
	at org.gradle.internal.graph.CachingDirectedGraphWalker.findValues(CachingDirectedGraphWalker.java:73)
	at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext.getDependencies(CachingTaskDependencyResolveContext.java:66)
	... 106 more


* Get more help at https://help.gradle.org

BUILD FAILED in 0s
19:59:03: Task execution finished 'publishReleaseBundle'.
```

--- a/plugin/build.gradle.kts
+++ b/plugin/build.gradle.kts
@@ -9,11 +9,11 @@ plugins {
 }
 
 dependencies {
-    compileOnly("com.android.tools.build:gradle:3.5.0-beta01")
+    compileOnly("com.android.tools.build:gradle:3.6.0-alpha02")
 
     implementation("com.google.apis:google-api-services-androidpublisher:v3-rev86-1.25.0")
 
-    testImplementation("com.android.tools.build:gradle:3.5.0-beta01")
+    testImplementation("com.android.tools.build:gradle:3.6.0-alpha02")
     testImplementation(kotlin("test"))
     testImplementation("junit:junit:4.12")
 }

--- a/plugin/src/main/kotlin/com/github/triplet/gradle/play/PlayPublisherExtension.kt
+++ b/plugin/src/main/kotlin/com/github/triplet/gradle/play/PlayPublisherExtension.kt
@@ -152,7 +152,7 @@ open class PlayPublisherExtension @JvmOverloads constructor(
             }
         }
 
-    @get:Internal("ProcessPackageMetadata is always out-of-date. Also, Closures with " +
+    @get:Internal("ProcessArtifactMetadata is always out-of-date. Also, Closures with " +
                           "parameters cannot be used as inputs.")
     internal var _outputProcessor: Action<ApkVariantOutput>? = null
 

--- a/plugin/src/main/kotlin/com/github/triplet/gradle/play/PlayPublisherPlugin.kt
+++ b/plugin/src/main/kotlin/com/github/triplet/gradle/play/PlayPublisherPlugin.kt
@@ -10,7 +10,7 @@ import com.github.triplet.gradle.play.internal.newTask
 import com.github.triplet.gradle.play.internal.validateRuntime
 import com.github.triplet.gradle.play.tasks.Bootstrap
 import com.github.triplet.gradle.play.tasks.GenerateResources
-import com.github.triplet.gradle.play.tasks.ProcessPackageMetadata
+import com.github.triplet.gradle.play.tasks.ProcessArtifactMetadata
 import com.github.triplet.gradle.play.tasks.PromoteRelease
 import com.github.triplet.gradle.play.tasks.PublishApk
 import com.github.triplet.gradle.play.tasks.PublishBundle
@@ -186,18 +186,18 @@ class PlayPublisherPlugin : Plugin<Project> {
             }
             publishProductsAllTask { dependsOn(publishProductsTask) }
 
-            val processPackageMetadata = project.newTask<ProcessPackageMetadata>(
+            val processArtifactMetadata = project.newTask<ProcessArtifactMetadata>(
                     "process${variantName}Metadata",
                     constructorArgs = arrayOf(extension, this)
             )
-            checkManifestProvider { dependsOn(processPackageMetadata) }
-            generateBuildConfigProvider { dependsOn(processPackageMetadata) }
+            checkManifestProvider { dependsOn(processArtifactMetadata) }
+            generateBuildConfigProvider { dependsOn(processArtifactMetadata) }
 
             val publishApkTaskDependenciesHack = project.newTask(
                     "publish${variantName}ApkWrapper"
             ) {
                 if (extension._artifactDir == null) {
-                    dependsOn(processPackageMetadata)
+                    dependsOn(processArtifactMetadata)
                     assembleProvider?.let {
                         dependsOn(it)
                     } ?: logger.warn("Assemble task not found. Publishing APKs may not work.")
@@ -231,7 +231,7 @@ class PlayPublisherPlugin : Plugin<Project> {
                     "publish${variantName}BundleWrapper"
             ) {
                 if (extension._artifactDir == null) {
-                    dependsOn(processPackageMetadata)
+                    dependsOn(processArtifactMetadata)
                     // TODO https://issuetracker.google.com/issues/109918868
                     project.tasks.findByName(
                             (this@whenObjectAdded as InstallableVariantImpl).variantData

--- a/plugin/src/main/kotlin/com/github/triplet/gradle/play/tasks/ProcessArtifactMetadata.kt
+++ b/plugin/src/main/kotlin/com/github/triplet/gradle/play/tasks/ProcessArtifactMetadata.kt
@@ -9,7 +9,7 @@ import com.github.triplet.gradle.play.tasks.internal.PlayPublishTaskBase
 import org.gradle.api.tasks.TaskAction
 import javax.inject.Inject
 
-open class ProcessPackageMetadata @Inject constructor(
+open class ProcessArtifactMetadata @Inject constructor(
         extension: PlayPublisherExtension,
         variant: ApplicationVariant
 ) : PlayPublishTaskBase(extension, variant) {

--- a/plugin/src/main/kotlin/com/github/triplet/gradle/play/tasks/PromoteRelease.kt
+++ b/plugin/src/main/kotlin/com/github/triplet/gradle/play/tasks/PromoteRelease.kt
@@ -3,7 +3,7 @@ package com.github.triplet.gradle.play.tasks
 import com.android.build.gradle.api.ApplicationVariant
 import com.github.triplet.gradle.play.PlayPublisherExtension
 import com.github.triplet.gradle.play.tasks.internal.ArtifactWorkerBase
-import com.github.triplet.gradle.play.tasks.internal.PlayPublishPackageBase
+import com.github.triplet.gradle.play.tasks.internal.PlayPublishArtifactBase
 import com.github.triplet.gradle.play.tasks.internal.UpdatableTrackExtensionOptions
 import com.github.triplet.gradle.play.tasks.internal.paramsForBase
 import org.gradle.api.tasks.Nested
@@ -17,7 +17,7 @@ import javax.inject.Inject
 open class PromoteRelease @Inject constructor(
         @get:Nested override val extension: PlayPublisherExtension,
         variant: ApplicationVariant
-) : PlayPublishPackageBase(extension, variant), UpdatableTrackExtensionOptions {
+) : PlayPublishArtifactBase(extension, variant), UpdatableTrackExtensionOptions {
     init {
         // Always out-of-date since we don't know what's changed on the network
         outputs.upToDateWhen { false }

--- a/plugin/src/main/kotlin/com/github/triplet/gradle/play/tasks/PublishApk.kt
+++ b/plugin/src/main/kotlin/com/github/triplet/gradle/play/tasks/PublishApk.kt
@@ -8,7 +8,7 @@ import com.github.triplet.gradle.play.internal.orNull
 import com.github.triplet.gradle.play.internal.playPath
 import com.github.triplet.gradle.play.internal.trackUploadProgress
 import com.github.triplet.gradle.play.tasks.internal.ArtifactWorkerBase
-import com.github.triplet.gradle.play.tasks.internal.PlayPublishPackageBase
+import com.github.triplet.gradle.play.tasks.internal.PlayPublishArtifactBase
 import com.github.triplet.gradle.play.tasks.internal.PublishableTrackExtensionOptions
 import com.github.triplet.gradle.play.tasks.internal.paramsForBase
 import com.google.api.client.googleapis.json.GoogleJsonResponseException
@@ -30,7 +30,7 @@ import javax.inject.Inject
 open class PublishApk @Inject constructor(
         @get:Nested override val extension: PlayPublisherExtension,
         variant: ApplicationVariant
-) : PlayPublishPackageBase(extension, variant), PublishableTrackExtensionOptions {
+) : PlayPublishArtifactBase(extension, variant), PublishableTrackExtensionOptions {
     @Suppress("MemberVisibilityCanBePrivate", "unused") // Used by Gradle
     @get:PathSensitive(PathSensitivity.RELATIVE)
     @get:InputFiles
@@ -79,7 +79,7 @@ open class PublishApk @Inject constructor(
                 return handleUploadFailures(e, content.file)
             }
 
-            handlePackageDetails(editId, apk.versionCode)
+            handleArtifactDetails(editId, apk.versionCode)
 
             return apk
         }

--- a/plugin/src/main/kotlin/com/github/triplet/gradle/play/tasks/PublishBundle.kt
+++ b/plugin/src/main/kotlin/com/github/triplet/gradle/play/tasks/PublishBundle.kt
@@ -1,21 +1,16 @@
 package com.github.triplet.gradle.play.tasks
 
 import com.android.build.gradle.api.ApplicationVariant
-import com.android.build.gradle.internal.api.InstallableVariantImpl
-import com.android.build.gradle.internal.scope.InternalArtifactType
 import com.github.triplet.gradle.play.PlayPublisherExtension
 import com.github.triplet.gradle.play.internal.MIME_TYPE_STREAM
 import com.github.triplet.gradle.play.internal.orNull
 import com.github.triplet.gradle.play.internal.playPath
 import com.github.triplet.gradle.play.internal.trackUploadProgress
 import com.github.triplet.gradle.play.tasks.internal.ArtifactWorkerBase
-import com.github.triplet.gradle.play.tasks.internal.PlayPublishPackageBase
-import com.github.triplet.gradle.play.tasks.internal.PublishableTrackExtensionOptions
+import com.github.triplet.gradle.play.tasks.internal.PublishBundleBase
 import com.github.triplet.gradle.play.tasks.internal.paramsForBase
 import com.google.api.client.googleapis.json.GoogleJsonResponseException
 import com.google.api.client.http.FileContent
-import org.gradle.api.tasks.InputFile
-import org.gradle.api.tasks.Nested
 import org.gradle.api.tasks.OutputDirectory
 import org.gradle.api.tasks.PathSensitive
 import org.gradle.api.tasks.PathSensitivity
@@ -28,25 +23,9 @@ import java.io.Serializable
 import javax.inject.Inject
 
 open class PublishBundle @Inject constructor(
-        @get:Nested override val extension: PlayPublisherExtension,
+        extension: PlayPublisherExtension,
         variant: ApplicationVariant
-) : PlayPublishPackageBase(extension, variant), PublishableTrackExtensionOptions {
-    @Suppress("MemberVisibilityCanBePrivate", "unused") // Used by Gradle
-    @get:PathSensitive(PathSensitivity.RELATIVE)
-    @get:InputFile
-    protected val bundle: File?
-        get() {
-            val customDir = extension._artifactDir
-
-            return if (customDir == null) {
-                (variant as InstallableVariantImpl).getFinalArtifact(InternalArtifactType.BUNDLE)
-                        .files.singleOrNull()
-            } else {
-                customDir.listFiles().orEmpty().singleOrNull { it.extension == "aab" }.also {
-                    if (it == null) println("Warning: no App Bundle found in '$customDir' yet.")
-                }
-            }
-        }
+) : PublishBundleBase(extension, variant) {
     @Suppress("MemberVisibilityCanBePrivate", "unused") // Used by Gradle
     @get:PathSensitive(PathSensitivity.RELATIVE)
     @get:OutputDirectory // This directory isn't used, but it's needed for up-to-date checks to work
@@ -75,7 +54,7 @@ open class PublishBundle @Inject constructor(
                 handleUploadFailures(e, content.file)
             } ?: return
 
-            handlePackageDetails(editId, bundle.versionCode)
+            handleArtifactDetails(editId, bundle.versionCode)
             updateTracks(editId, listOf(bundle.versionCode.toLong()))
         }
 

--- a/plugin/src/main/kotlin/com/github/triplet/gradle/play/tasks/PublishInternalSharingBundle.kt
+++ b/plugin/src/main/kotlin/com/github/triplet/gradle/play/tasks/PublishInternalSharingBundle.kt
@@ -1,19 +1,14 @@
 package com.github.triplet.gradle.play.tasks
 
 import com.android.build.gradle.api.ApplicationVariant
-import com.android.build.gradle.internal.api.InstallableVariantImpl
-import com.android.build.gradle.internal.scope.InternalArtifactType
 import com.github.triplet.gradle.play.PlayPublisherExtension
 import com.github.triplet.gradle.play.internal.MIME_TYPE_STREAM
 import com.github.triplet.gradle.play.internal.orNull
 import com.github.triplet.gradle.play.internal.trackUploadProgress
-import com.github.triplet.gradle.play.tasks.internal.ArtifactExtensionOptions
-import com.github.triplet.gradle.play.tasks.internal.PlayPublishTaskBase
 import com.github.triplet.gradle.play.tasks.internal.PlayWorkerBase
+import com.github.triplet.gradle.play.tasks.internal.PublishBundleBase
 import com.github.triplet.gradle.play.tasks.internal.paramsForBase
 import com.google.api.client.http.FileContent
-import org.gradle.api.tasks.InputFile
-import org.gradle.api.tasks.Nested
 import org.gradle.api.tasks.OutputDirectory
 import org.gradle.api.tasks.PathSensitive
 import org.gradle.api.tasks.PathSensitivity
@@ -26,25 +21,9 @@ import java.io.Serializable
 import javax.inject.Inject
 
 open class PublishInternalSharingBundle @Inject constructor(
-        @get:Nested override val extension: PlayPublisherExtension,
+        extension: PlayPublisherExtension,
         variant: ApplicationVariant
-) : PlayPublishTaskBase(extension, variant), ArtifactExtensionOptions {
-    @Suppress("MemberVisibilityCanBePrivate", "unused") // Used by Gradle
-    @get:PathSensitive(PathSensitivity.RELATIVE)
-    @get:InputFile
-    protected val bundle: File?
-        get() {
-            val customDir = extension._artifactDir
-
-            return if (customDir == null) {
-                (variant as InstallableVariantImpl).getFinalArtifact(InternalArtifactType.BUNDLE)
-                        .files.singleOrNull()
-            } else {
-                customDir.listFiles().orEmpty().singleOrNull { it.extension == "aab" }.also {
-                    if (it == null) println("Warning: no App Bundle found in '$customDir' yet.")
-                }
-            }
-        }
+) : PublishBundleBase(extension, variant) {
     @Suppress("MemberVisibilityCanBePrivate", "unused") // Used by Gradle
     @get:PathSensitive(PathSensitivity.RELATIVE)
     @get:OutputDirectory

--- a/plugin/src/main/kotlin/com/github/triplet/gradle/play/tasks/internal/PlayPublishArtifactBase.kt
+++ b/plugin/src/main/kotlin/com/github/triplet/gradle/play/tasks/internal/PlayPublishArtifactBase.kt
@@ -13,7 +13,7 @@ import org.gradle.api.tasks.PathSensitive
 import org.gradle.api.tasks.PathSensitivity
 import java.io.File
 
-abstract class PlayPublishPackageBase(
+abstract class PlayPublishArtifactBase(
         extension: PlayPublisherExtension,
         variant: ApplicationVariant
 ) : PlayPublishTaskBase(extension, variant) {

--- a/plugin/src/main/kotlin/com/github/triplet/gradle/play/tasks/internal/PlayWorkers.kt
+++ b/plugin/src/main/kotlin/com/github/triplet/gradle/play/tasks/internal/PlayWorkers.kt
@@ -36,7 +36,7 @@ internal fun PlayPublishTaskBase.paramsForBase(
             editId
     )
 
-    if (this is PlayPublishPackageBase) {
+    if (this is PlayPublishArtifactBase) {
         val artifact = ArtifactWorkerBase.ArtifactPublishingData(
                 variant.name,
                 variant.outputs.map { it.versionCode }.first(),
@@ -194,7 +194,7 @@ internal abstract class ArtifactWorkerBase(
         throw e
     }
 
-    protected fun handlePackageDetails(editId: String, versionCode: Int) {
+    protected fun handleArtifactDetails(editId: String, versionCode: Int) {
         val file = artifact.mappingFile
         if (file != null && file.length() > 0) {
             val mapping = FileContent(MIME_TYPE_STREAM, file)

--- a/plugin/src/main/kotlin/com/github/triplet/gradle/play/tasks/internal/PublishBundleBase.kt
+++ b/plugin/src/main/kotlin/com/github/triplet/gradle/play/tasks/internal/PublishBundleBase.kt
@@ -1,0 +1,39 @@
+package com.github.triplet.gradle.play.tasks.internal
+
+import com.android.build.gradle.api.ApplicationVariant
+import com.android.build.gradle.internal.api.InstallableVariantImpl
+import com.android.build.gradle.internal.scope.InternalArtifactType
+import com.github.triplet.gradle.play.PlayPublisherExtension
+import com.github.triplet.gradle.play.internal.orNull
+import org.gradle.api.file.RegularFile
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.Nested
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.PathSensitivity
+import java.io.File
+import javax.inject.Inject
+
+open class PublishBundleBase @Inject constructor(
+        @get:Nested override val extension: PlayPublisherExtension,
+        variant: ApplicationVariant
+) : PlayPublishArtifactBase(extension, variant), PublishableTrackExtensionOptions {
+    @Suppress("MemberVisibilityCanBePrivate", "unused") // Used by Gradle
+    @get:PathSensitive(PathSensitivity.RELATIVE)
+    @get:InputFile
+    protected val bundle: File?
+        get() {
+            val customDir = extension._artifactDir
+
+            return if (customDir == null) {
+                val installable = variant as InstallableVariantImpl
+                installable.variantData.scope.artifacts
+                        .getFinalProduct<RegularFile>(InternalArtifactType.BUNDLE)
+                        .get().asFile.orNull() ?: installable
+                        .getFinalArtifact(InternalArtifactType.BUNDLE).files.singleOrNull()
+            } else {
+                customDir.listFiles().orEmpty().singleOrNull { it.extension == "aab" }.also {
+                    if (it == null) println("Warning: no App Bundle found in '$customDir' yet.")
+                }
+            }
+        }
+}

--- a/plugin/src/test/groovy/com/github/triplet/gradle/play/TestHelper.groovy
+++ b/plugin/src/test/groovy/com/github/triplet/gradle/play/TestHelper.groovy
@@ -61,7 +61,7 @@ class TestHelper {
                 jcenter()
             }
             dependencies {
-                classpath 'com.android.tools.build:gradle:3.5.0-beta01'
+                classpath 'com.android.tools.build:gradle:3.6.0-alpha02'
                 classpath files('../../../../build/libs/plugin-${System.getProperty("VERSION_NAME")}.jar')
 
                 // Manually define transitive dependencies for our plugin since we don't have the

--- a/plugin/src/test/kotlin/com/github/triplet/gradle/play/CompatibilityTest.kt
+++ b/plugin/src/test/kotlin/com/github/triplet/gradle/play/CompatibilityTest.kt
@@ -103,7 +103,7 @@ class CompatibilityTest(
         fun parameters() = listOf(
                 arrayOf("3.5.0-beta01", "5.4"), // Oldest supported
                 arrayOf("3.5.0-beta01", "5.4"), // Latest stable
-                arrayOf("3.5.0-beta01", "5.4") // Latest
+                arrayOf("3.6.0-alpha02", "5.4.1") // Latest
         )
     }
 }

--- a/plugin/src/test/kotlin/com/github/triplet/gradle/play/Helpers.kt
+++ b/plugin/src/test/kotlin/com/github/triplet/gradle/play/Helpers.kt
@@ -20,7 +20,7 @@ fun execute(androidConfig: String, expectFailure: Boolean, vararg tasks: String)
                 jcenter()
             }
             dependencies {
-                classpath 'com.android.tools.build:gradle:3.5.0-beta01'
+                classpath 'com.android.tools.build:gradle:3.6.0-alpha02'
                 classpath files('../../../../build/libs/plugin-${System.getProperty("VERSION_NAME")}.jar')
 
                 // Manually define transitive dependencies for our plugin since we don't have the

